src/program/snabbvmx/doc/README.userguide.md

```diff
@@ -0,0 +1,368 @@
+lwaftr_v6 out-ipv6-frag                                  0                    0
+lwaftr_v6 out-ipv6-frag-not                              0                    0
+
+                                                     Total           per second
+nic ifInDiscards                                 1,547,333               47,449
+```
+
+1.53 MPPS in both interfaces rx/tx. We are running at line-rate, although there
+ are packet drops.
+
+## SnabbVMX query
+
+SnabbVMX query prints out all the counters of a SnabbVMX instance in XML format:
+
+```xml
+<snabb>
+  <instance>
+   <id>0</id>
+   <name>xe0</name>
+   <pid>11958</pid>
+   <next_hop_mac_v4>00:00:00:00:00:00</next_hop_mac_v4>
+   <next_hop_mac_v6>00:00:00:00:00:00</next_hop_mac_v6>
+   <monitor>0.0.0.0</monitor>
+   <engine>
+     <breaths>67198200</breaths>
+     <configs>1</configs>
+     <freebits>4663620304544</freebits>
+     <freebytes>569220658720</freebytes>
+     <frees>1525764372</frees>
+   </engine>
+   <pci>
+   ...
+   </pci>
+   <apps>
+     <lwaftr>
+       <drop-all-ipv4-iface-bytes>4488858</drop-all-ipv4-iface-bytes>
+       <drop-all-ipv4-iface-packets>12721</drop-all-ipv4-iface-packets>
+       <drop-all-ipv6-iface-bytes>4997698</drop-all-ipv6-iface-bytes>
+       <in-ipv4-bytes>269348948180</in-ipv4-bytes>
+       <in-ipv4-frag-needs-reassembly>0</in-ipv4-frag-needs-reassembly>
+       <in-ipv4-frag-reassembled>0</in-ipv4-frag-reassembled>
+       <in-ipv4-frag-reassembly-unneeded>0</in-ipv4-frag-reassembly-unneeded>
+     </lwaftr>
+   </apps>
+   <links>
+     <vm_v4v6.v4----nh_fwd4.vm>
+       <dtime>1476276089</dtime>
+       <rxbytes>0</rxbytes>
+       <rxpackets>0</rxpackets>
+       <txbytes>0</txbytes>
+       <txdrop>0</txdrop>
+       <txpackets>0</txpackets>
+     </vm_v4v6.v4----nh_fwd4.vm>
+   </links>
+  </instance>
+</snabb>
+```
+
+Useful for interacting with other systems in the network.
+
+Snabb's lwAFTR features a query tool too. This tool is aimed to query lwAFTR's
+counters.  Snabb's `lwaftr query` is covered in the last section of this chapter.
+
+## SnabbVMX check
+
+`snabbvmx check` is an utility tool use to validate lwAFTR correctnes.
+SnabbVMX has its own version of it.
+
+```bash
+$ sudo ./snabb snabbvmx check
+Usage: check [-r] CONF V4-IN.PCAP V6-IN.PCAP V4-OUT.PCAP V6-OUT.PCAP
+[COUNTERS.LUA]
+```
+
+Using check is the step prior to add a new test in the end-to-end test suite.
+
+**CONF** is a SnabbVMX configuration file and **V4-IN** and **V6-IN** are the
+incoming V4 and V6 interfaces. The tools reads packets from a pcap file. **V4-OUT**
+and **V6-OUT** are the resulting packets after the lwAFTR processing. How the
+packets are processed depends on the configuration file and the binding table.
+
+Although SnabbVMX works in one single interface, snabbvmx check requires that
+the packet split is already done and provides an split output too.
+
+Imagine you have detected an error in the lwAFTR. First step is to obtain the
```

Comment:
The text from here to the end of this chapter is the same as the **How to write a SnabbVMX end-to-end test** chapter at the end of `README.troubleshooting.md`. Is that intended?